src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,644 @@
+        // 3 bits.
+        //
+        // This gate also checks that this window uses the correct y_p and
+        // interpolated x_p.
+        meta.create_gate("Decompose base field element", |meta| {
+            let base_field_fixed_mul = meta.query_selector(self.base_field_fixed_mul);
+
+            let z_cur = meta.query_advice(self.super_config.window, Rotation::cur());
+            let z_next = meta.query_advice(self.super_config.window, Rotation::next());
+
+            //    z_{i+1} = (z_i - a_i) / (2^3)
+            // => a_i = z_i - (z_{i+1} * (2^3))
+            let word = z_cur - z_next * pallas::Base::from_u64(constants::H as u64);
+
+            // (word - 7) * (word - 6) * ... * (word - 1) * word = 0
+            let range_check = range_check(word.clone(), constants::H);
+
+            self.super_config
+                .coords_check(meta, base_field_fixed_mul.clone(), word)
+                .into_iter()
+                .chain(Some((
+                    "Decomposition range check",
+                    base_field_fixed_mul * range_check,
+                )))
+        });
+
+        // Check that we get z_85 = 0 as the final output of the three-bit decomposition running sum.
+        // Also check that the base field element is canonical.
+        meta.create_gate("Canonicity checks", |meta| {
+            let base_field_fixed_canon = meta.query_selector(self.base_field_fixed_canon);
+
+            let alpha = meta.query_advice(self.canon_advices[0], Rotation::prev());
+            // z_85_alpha is constrained to be zero in this gate.
+            let z_85_alpha = meta.query_advice(self.canon_advices[1], Rotation::prev());
+            // The last three bits of α.
+            let z_84_alpha = meta.query_advice(self.canon_advices[2], Rotation::prev());
+
+            // Decompose α into three pieces,
+            //            α = α_0 (252 bits)  || α_1 (2 bits) || α_2 (1 bit).
+            //
+            let alpha_0 = meta.query_advice(self.canon_advices[0], Rotation::cur());
+            let alpha_1 = meta.query_advice(self.canon_advices[1], Rotation::cur());
+            let alpha_2 = meta.query_advice(self.canon_advices[2], Rotation::cur());
+
+            let alpha_0_prime = meta.query_advice(self.canon_advices[0], Rotation::next());
+            let z_13_alpha_0_prime = meta.query_advice(self.canon_advices[1], Rotation::next());
+            let z_13_alpha_0 = meta.query_advice(self.canon_advices[2], Rotation::next());
+
+            let decomposition_checks = {
+                // Range-constrain α_1 to be 2 bits
+                let alpha_1_range_check = range_check(alpha_1.clone(), 1 << 2);
+                // Boolean-constrain α_2
+                let alpha_2_range_check = range_check(alpha_2.clone(), 1 << 1);
+                // Check that α_1 + 2^2 α_2 = z_84_alpha
+                let z_84_alpha_check = z_84_alpha.clone()
+                    - (alpha_1 + alpha_2.clone() * pallas::Base::from_u64(1 << 2));
```

Comment:
```suggestion
                    - (alpha_1.clone() + alpha_2.clone() * pallas::Base::from_u64(1 << 2));
```
(we end up needing `alpha_1` again below).